update upgrade notes with CRD schema instructions (#5174)
istio/istio.io: content/en/news/2019/announcing-1.4/upgrade-notes/index.md

* Specify the environment variable `PILOT_BLOCK_HTTP_ON_443=false` to the Pilot deployment. Note: this may be removed in future releases.

See [Protocol Selection](/docs/ops/traffic-management/protocol-selection/) for more information about specifying the protocol of a port

> ## Configuration Management

Reviewer: ```suggestion
## Configuration management
```

Author: I thought we are using capitals for all words in a title? like above Traffic Management

Reviewer: Nope, only for doc titles. Headings use first capital only.

https://istio.io/about/contribute/style-guide/#use-title-capitalization-for-title-front-matter

Author: ok. Done. Also changed the traffic management heading.

> We introduced OpenAPI v3 schemas in the Kubernetes [CustomResourceDefinitions(CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) of Istio resources. The schemas describe the Istio resources and help ensure the Istio resources you create and modify are structurally correct.

Reviewer: ```suggestion
We introduced OpenAPI v3 schemas in the Kubernetes [custom resource definitions (CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) of Istio resources. The schemas describe the Istio resources and help ensure the Istio resources you create and modify are structurally correct.
```

If one or more fields in your configurations are unknown or have wrong types, they will be rejected by the Kubernetes API server when you create or modify Istio resources. This feature, `CustomResourceValidation`, is on by default for Kubernetes 1.9+ clusters. Please note that existing configurations already in Kubernetes are __NOT__ affected if they stay unchanged.

To help with your upgrade, here are some steps you could take:

* After upgrading Istio, run your Istio configurations with `kubectl apply --dry-run` so that you are able to know if the configurations can be accepted by the API server as well as any possible unknown and/or invalid fields to the API server. (`DryRun` feature is on by default for Kubernetes 1.13+ clusters.)

> * Use the [reference docs](/docs/reference/config/) and/or `kubectl explain` (more details below) to confirm and correct the field names and data types.

Reviewer: ```suggestion
* Use the [reference documentation](/docs/reference/config/) and/or `kubectl explain` (more details below) to confirm and correct the field names and data types.
```

> If you choose to ignore the validation errors, add `--validate=false` to your `kubectl` command when you create or modify Istio resources, we do not, however, recommend doing this.

Reviewer: ```suggestion
If you choose to ignore the validation errors, add `--validate=false` to your `kubectl` command when you create or modify Istio resources. We strongly discourage doing so however, since it is willingly introducing incorrect configuration.
```
